dms/models/dms_file.py

```diff
@@ -643,6 +699,13 @@ def unlink(self):
         # will be deleted
         return super(File, self.sudo()).unlink()
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if "res_model" not in vals and "res_id" not in vals:
+                vals = self._create_model_attachment(vals)
+        return super(File, self).create(vals_list)
```

Comment:
This is not doing the changes, as the new `vals` is not on `vals_list`. You have to build a new vals_list:
```suggestion
        new_vals_list = []
        for vals in vals_list:
            if "res_model" not in vals and "res_id" not in vals:
                vals = self._create_model_attachment(vals)
            new_vals_list.append(vals)
        return super(File, self).create(new_vals_list)
```